cmd/github-labels/github-labels.go

```diff
@@ -0,0 +1,683 @@
+	case defaultFormat:
+		fmt.Printf("    %s (%q) [category %q, colour %q, from %q]\n", l.Name, desc, l.CategoryName, l.Colour, l.From)
+	case githubFormat:
+		var fromText string
+
+		if l.From != "" {
+			fromText = fmt.Sprintf("`%s`", l.From)
+
+		}
+		fmt.Printf("%c`%s`%c%s%c`%s`%c`%s`%c%s%c\n",
+			githubTableSeparator,
+			l.Name,
+			githubTableSeparator,
+			desc,
+			githubTableSeparator,
+			l.CategoryName,
+			githubTableSeparator,
+			l.Colour,
+			githubTableSeparator,
+			fromText,
+			githubTableSeparator)
+	case tsvFormat:
+		fmt.Printf("%s\t%s\t%s\t%s\t%s\t\n",
+			l.Name, desc, l.CategoryName, l.Colour, l.From)
+	}
+}
+
+func showLabels(lf *LabelsFile, format Format) {
+	switch format {
+	case defaultFormat:
+		fmt.Printf("Labels (count: %d):\n", len(lf.Labels))
+	case githubFormat:
+		fmt.Printf("%c%s%c%s%c%s%c%s%c%s%c\n",
+			githubTableSeparator,
+			"Name",
+			githubTableSeparator,
+			"Description",
+			githubTableSeparator,
+			"Category",
+			githubTableSeparator,
+			"Colour",
+			githubTableSeparator,
+			"From",
+			githubTableSeparator)
+		fmt.Printf("|-|-|-|-|-|\n")
+	case tsvFormat:
+		fmt.Printf("%s\t%s\t%s\t%s\t%s\t\n",
+			"Name",
+			"Description",
+			"Category",
+			"Colour",
+			"From")
+	}
+
+	for _, l := range lf.Labels {
+		showLabel(l, format)
+	}
+}
+
+func getLabelsByCategory(categoryName string, lf *LabelsFile) ([]Label, error) {
+	var labels []Label
+
+	if categoryName == "" {
+		return nil, errors.New("need category name")
+	}
+
+	for _, label := range lf.Labels {
+		if label.CategoryName == categoryName {
+			labels = append(labels, label)
+		}
+	}
+
+	return labels, nil
+}
+
+func showCategory(c Category, lf *LabelsFile, showLabels bool, format Format) error {
+	desc := strings.Replace(c.Description, "\n", " ", -1)
+	desc = strings.TrimSpace(desc)
+
+	switch format {
+	case defaultFormat:
+		/* NOP */
+	case githubFormat:
+		fmt.Printf("%c`%s`%c%s%c%s%c\n",
+			githubTableSeparator,
+			c.Name,
+			githubTableSeparator,
+			desc,
+			githubTableSeparator,
+			c.URL,
+			githubTableSeparator)
+	case tsvFormat:
+		fmt.Printf("%s\t%s\t%s\t\n",
+			c.Name, desc, c.URL)
+	}
+
+	if format != defaultFormat {
+		return nil
+	}
+
+	if showLabels {
+		labels, err := getLabelsByCategory(c.Name, lf)
+		if err != nil {
+			return err
+		}
+
+		fmt.Printf("    %s (%q, label count: %d)\n", c.Name, desc, len(labels))
+
+		for _, label := range labels {
+			fmt.Printf("        %s (%q)\n", label.Name, label.Description)
+		}
+	} else {
+		fmt.Printf("    %s (%q)\n", c.Name, desc)
+	}
+
+	return nil
+}
+
+func showCategories(lf *LabelsFile, showLabels bool, format Format) error {
+	switch format {
+	case defaultFormat:
+		fmt.Printf("Categories (count: %d):\n", len(lf.Categories))
+	case githubFormat:
+		fmt.Printf("%c%s%c%s%c%s%c\n",
+			githubTableSeparator,
+			"Name",
+			githubTableSeparator,
+			"Description",
+			githubTableSeparator,
+			"URL",
+			githubTableSeparator)
+		fmt.Printf("|-|-|-|\n")
```

Comment:
Don't know if it will help or not, but for the checkmetrics tool table results I used https://github.com/olekukonko/tablewriter#example-5---markdown-format - and it looks like it can do markdown output with some tweaks.

Reply:
It might be overkill tbh, but I'll take a look...